crates/bdk/src/wallet/mod.rs

```diff
@@ -1986,6 +1986,49 @@ impl<D> Wallet<D> {
     pub fn local_chain(&self) -> &LocalChain {
         &self.chain
     }
+
+    /// Set lookahead for all keychains.
+    pub fn set_lookahead_for_all(&mut self, lookahead: u32) {
+        // Having a lookahead of 0, means if we sync the chain
+        // we would not find any update since we don't have any
+        // store scripts in our indexer.
+        assert_ne!(lookahead, 0, "lookahead must be greater than 0");
+        self.indexed_graph.index.set_lookahead_for_all(lookahead);
+    }
```

Comment:
Is there a nicer way to handle the case when `lookahead` is 0 besides panicking? If not, it should be mentioned in the function's doc comment, but we might also consider returning a `Result`, or setting some default value for lookahead.

Reply:
I think returning an error here is better. I have also added more comments about this in the docs.